### ISSUE TITLE
Huge table makes application (or JavaVM?) run out of Java heap space

### ISSUE THREAD

**Issue by larsarus** [keeps/db-preservation-toolkit#318]
Description: 

Steps required to reproduce the bug:

1. Have database with huge table, i.e. 70 million rows / 26 GB.
2. Run DBPTK (even with various increases in max heap space and GC tuning)
3. DBPTK / JavaVM uses up all available memory, then crashes.

I have a machine with 32 GB RAM, and allowing the app to use all of it still isn't enough. The problem exists independently of the database server. Maybe there needs to be some code to detect huge tables and process them in a special way to not run out of resources.

Attach the dbptk-app.log.txt file below.

I forgot to save the log-file, but will attach it once I the re-run of the test is finished.

**Comment by larsarus** [comment]
Logs and screenshot. It now shows "GC overhead limit exceeded" but the problem is identical.
[dbptk-logs.zip](https://github.com/keeps/db-preservation-toolkit/files/1104671/dbptk-logs.zip)
![gc_overload](https://user-images.githubusercontent.com/22052458/27578444-610baba2-5b24-11e7-800a-bcc9579d1e29.PNG)

**Comment by larsarus** [comment]
I can upload the big table if you want it for debugging. It contains no sensitive data.

**Comment by chalkos** [comment]
It would be nice to have the big table for debug. Or the whole database if you can upload it.

**Comment by chalkos** [comment]
@larsarus I could not find anything suspicious from the logs. Are you still able to provide me the table/database?

**Comment by larsarus** [comment]
I'm sorry, I do not. I'll check if someone else it able to provide you with the table or relevant test-data.

**Comment by luis100** [comment]
This may be due to the specific database driver being used, the options defined (or not defined) for the driver, together with particularities of the database being ingested. We have made thorough tests to reduce the processing and memory needed, but some drivers do a lot of caching in memory or, for example, put a whole blob/row/resultset in memory when fetching. But, some drivers also provide other methods to configure of fetch information that do not use so much memory (defining cursor fetching configuration, specific driver methods to stream blobs, etc.

So we would need more information to be able to reproduce the problem and find a solution, specifically test-data and a specific DBMS where this behaviour is consistent.

Closing this issue while this information cannot be provided.
